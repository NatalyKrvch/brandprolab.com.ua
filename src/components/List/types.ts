export interface ListProps {
  list: Array<string>;
}
