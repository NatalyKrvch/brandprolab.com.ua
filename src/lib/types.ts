export type Size = 's' | 'm' | 'l';
export type Screen = 'mobile' | 'tablet' | 'desktop';
