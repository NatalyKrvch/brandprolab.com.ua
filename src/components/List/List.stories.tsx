import type { Meta, StoryObj } from '@storybook/react';

import List from './List';

const meta: Meta<typeof List> = {
  title: 'Components/List',
  component: List,
  parameters: {
    controls: { expanded: true },
  },
  argTypes: {
    list: {
      control: {
        type: 'object',
      },
      description: 'An array of strings representing list items.',
      table: {
        type: { summary: 'string[]' },
      },
    },
  },
};

export default meta;
type Story = StoryObj<typeof List>;

export const Default: Story = {
  args: {
    list: ['High quality services', 'Affordable pricing', 'Friendly support'],
  },
};

export const CustomizableList: Story = {
  args: {
    list: ['Custom item 1', 'Custom item 2'],
  },
};
