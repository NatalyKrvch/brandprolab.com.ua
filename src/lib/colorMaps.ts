import {
  MainSocialCardColor,
  socialMediaColorClassMapType,
} from '@/components/Cards/SocialMediaCard/types';

export const socialMediaColorClassMap: socialMediaColorClassMapType = {
  [MainSocialCardColor.RED]: {
    bg: 'bg-red-default',
    text: 'text-red-default',
  },
  [MainSocialCardColor.BLUE]: {
    bg: 'bg-blue-default',
    text: 'text-blue-default',
  },
};
