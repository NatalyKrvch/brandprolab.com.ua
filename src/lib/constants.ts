export const ALL_DATA_QUERY = `
{
  "header": *[_type == "header"][0],
  "hero": *[_type == "hero"][0],
  "cases": *[_type == "cases"][0],
  "services": *[_type == "services"][0],
  "testimonials": *[_type == "testimonialsSection"][0],
  "subscribe": *[_type == "subscribeSection"][0],
  "video": *[_type == "videoSection"][0],
  "expertise": *[_type == "expertiseSection"][0],
  "diagnostic": *[_type == "diagnosticSection"][0],
  "footer": *[_type == "footerSection"][0]
}
`;

export const YOUTUBE_REGEX =
  /(?:youtu\.be\/|youtube\.com\/(?:.*[?&]v=|embed\/|v\/|shorts\/|live\/))([^&?/]+)/;

export enum IconClassType {
  SERVICES = 'services',
  EXPERTISE = 'expertise',
  SOCIAL_MEDIA = 'social-media',
  SOCIAL_MEDIA_MAIN = 'social-media-main',
}
export const REVALIDATE = 60;
export const ENDLESS_CAROUSEL_SPEED = 50;
export const SLIDER_RIGHT_SHIFT = 2;
export const SLIDER_LEFT_SHIFT = -1;
export const AMOUNT_OF_TESTIMONIALS_WORDS = 15;
export const AMOUNT_OF_CASES_WORDS_DESKTOP = 13;
export const AMOUNT_OF_CASES_WORDS_TABLET = 9;
export const AMOUNT_OF_CASES_WORDS_MOBILE = 13;
