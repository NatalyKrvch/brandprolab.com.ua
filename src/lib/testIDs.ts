export const INTERNAL_LINK_TEST_ID = 'internal-link-test-id';
export const EXTERNAL_LINK_TEST_ID = 'external-link-test-id';
export const FULL_WIDTH_CONTAINER_TEST_ID = 'full-width-container-test-id';
export const LIMITED_WIDTH_CONTAINER_TEST_ID =
  'limited-width-container-test-id';
export const ICON_COMPONENT_TEST_ID = 'icon-component-test-id';
export const ICON_COMPONENT_IMAGE_TEST_ID = 'icon-component-image-test-id';
export const LABEL_COMPONENT_TEST_ID = 'label-component-test-id';
export const BUTTON_TEST_ID = 'button-test-id';
export const UNDERLINED_TEXT_TEST_ID = 'underlined-text-test-id';
export const USER_BADGE_TEST_ID = 'user-badge-test-id';
export const USER_BADGE_IMAGE_TEST_ID = 'user-badge-image-test-id';
export const USER_BADGE_NAME_TEST_ID = 'user-badge-name-test-id';
export const TESTIMONIAL_CARD_TEST_ID = 'testimonials-card-test-id';
export const TESTIMONIAL_TEXT_TEST_ID = 'testimonials-text-test-id';
export const LIST_TEST_ID = 'list-test-id';
export const LIST_ITEM_TEST_ID = 'list-item-test-id';
