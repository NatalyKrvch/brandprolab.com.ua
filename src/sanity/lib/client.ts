import { createClient } from 'next-sanity';

import { getSanityConfig } from '../env';

const { apiVersion, dataset, projectId, useCdn } = getSanityConfig();

export const client = createClient({
  projectId,
  dataset,
  apiVersion,
  useCdn,
});
