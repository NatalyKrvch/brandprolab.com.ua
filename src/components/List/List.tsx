import Image from 'next/image';

import { LIST_ITEM_TEST_ID, LIST_TEST_ID } from '@/lib/testIDs';

import {
  getCheckIconWrapperClass,
  getListItemClass,
  getListWrapperClass,
} from './helpers/getListClasses';
import type { ListProps } from './types';

const List = ({ list }: ListProps) => {
  return (
    <ul className={getListWrapperClass()} data-testid={LIST_TEST_ID}>
      {list.map(item => (
        <li
          key={item}
          className={getListItemClass()}
          data-testid={LIST_ITEM_TEST_ID}
        >
          <div className={getCheckIconWrapperClass()}>
            <Image
              src="/icons/cards/check.svg"
              fill
              alt="Checkmark icon"
              aria-hidden="true"
            />
          </div>
          {item}
        </li>
      ))}
    </ul>
  );
};

export default List;
