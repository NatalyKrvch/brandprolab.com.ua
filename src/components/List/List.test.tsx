import '@testing-library/jest-dom';

import { render, screen } from '@testing-library/react';

import { LIST_ITEM_TEST_ID, LIST_TEST_ID } from '@/lib/testIDs';

import List from './List';

jest.mock('next/image', () => {
  const MockImage = ({ src, alt }: { src: string; alt: string }) => (
    <img src={src} alt={alt} />
  );
  MockImage.displayName = 'MockImage';
  return MockImage;
});

describe('List component', () => {
  const defaultProps = {
    list: ['Item 1', 'Item 2', 'Item 3'],
  };

  it('renders the List component', () => {
    const { asFragment } = render(<List {...defaultProps} />);
    const listElement = screen.getByTestId(LIST_TEST_ID);

    expect(listElement).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders the correct number of list items', () => {
    const { asFragment } = render(<List {...defaultProps} />);
    const listItems = screen.getAllByTestId(LIST_ITEM_TEST_ID);

    expect(listItems).toHaveLength(defaultProps.list.length);
    expect(asFragment()).toMatchSnapshot();

    listItems.forEach((item, index) => {
      expect(item).toHaveTextContent(defaultProps.list[index]);
    });
  });

  it('renders checkmark images for each list item', () => {
    const { asFragment } = render(<List {...defaultProps} />);
    const listItems = screen.getAllByTestId(LIST_ITEM_TEST_ID);

    listItems.forEach(item => {
      const imageElement = item.querySelector('img');
      expect(imageElement).toBeInTheDocument();
      expect(imageElement).toHaveAttribute('src', '/icons/cards/check.svg');
    });
    expect(asFragment()).toMatchSnapshot();
  });
});
