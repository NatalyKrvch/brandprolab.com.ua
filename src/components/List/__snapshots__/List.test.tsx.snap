// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component renders checkmark images for each list item 1`] = `
<DocumentFragment>
  <ul
    class="flex flex-col gap-3 tablet:gap-4 desktop:gap-4"
    data-testid="list-test-id"
  >
    <li
      class="align-center flex gap-2 text-base tablet:gap-3 tablet:text-lg desktop:gap-4 desktop:text-lg"
      data-testid="list-item-test-id"
    >
      <div
        class="relative h-4 w-4 tablet:h-5 tablet:w-5 desktop:h-6 desktop:w-6"
      >
        <img
          alt=""
          src="/icons/cards/check.svg"
        />
      </div>
      Item 1
    </li>
    <li
      class="align-center flex gap-2 text-base tablet:gap-3 tablet:text-lg desktop:gap-4 desktop:text-lg"
      data-testid="list-item-test-id"
    >
      <div
        class="relative h-4 w-4 tablet:h-5 tablet:w-5 desktop:h-6 desktop:w-6"
      >
        <img
          alt=""
          src="/icons/cards/check.svg"
        />
      </div>
      Item 2
    </li>
    <li
      class="align-center flex gap-2 text-base tablet:gap-3 tablet:text-lg desktop:gap-4 desktop:text-lg"
      data-testid="list-item-test-id"
    >
      <div
        class="relative h-4 w-4 tablet:h-5 tablet:w-5 desktop:h-6 desktop:w-6"
      >
        <img
          alt=""
          src="/icons/cards/check.svg"
        />
      </div>
      Item 3
    </li>
  </ul>
</DocumentFragment>
`;

exports[`List component renders the List component 1`] = `
<DocumentFragment>
  <ul
    class="flex flex-col gap-3 tablet:gap-4 desktop:gap-4"
    data-testid="list-test-id"
  >
    <li
      class="align-center flex gap-2 text-base tablet:gap-3 tablet:text-lg desktop:gap-4 desktop:text-lg"
      data-testid="list-item-test-id"
    >
      <div
        class="relative h-4 w-4 tablet:h-5 tablet:w-5 desktop:h-6 desktop:w-6"
      >
        <img
          alt=""
          src="/icons/cards/check.svg"
        />
      </div>
      Item 1
    </li>
    <li
      class="align-center flex gap-2 text-base tablet:gap-3 tablet:text-lg desktop:gap-4 desktop:text-lg"
      data-testid="list-item-test-id"
    >
      <div
        class="relative h-4 w-4 tablet:h-5 tablet:w-5 desktop:h-6 desktop:w-6"
      >
        <img
          alt=""
          src="/icons/cards/check.svg"
        />
      </div>
      Item 2
    </li>
    <li
      class="align-center flex gap-2 text-base tablet:gap-3 tablet:text-lg desktop:gap-4 desktop:text-lg"
      data-testid="list-item-test-id"
    >
      <div
        class="relative h-4 w-4 tablet:h-5 tablet:w-5 desktop:h-6 desktop:w-6"
      >
        <img
          alt=""
          src="/icons/cards/check.svg"
        />
      </div>
      Item 3
    </li>
  </ul>
</DocumentFragment>
`;

exports[`List component renders the correct number of list items 1`] = `
<DocumentFragment>
  <ul
    class="flex flex-col gap-3 tablet:gap-4 desktop:gap-4"
    data-testid="list-test-id"
  >
    <li
      class="align-center flex gap-2 text-base tablet:gap-3 tablet:text-lg desktop:gap-4 desktop:text-lg"
      data-testid="list-item-test-id"
    >
      <div
        class="relative h-4 w-4 tablet:h-5 tablet:w-5 desktop:h-6 desktop:w-6"
      >
        <img
          alt=""
          src="/icons/cards/check.svg"
        />
      </div>
      Item 1
    </li>
    <li
      class="align-center flex gap-2 text-base tablet:gap-3 tablet:text-lg desktop:gap-4 desktop:text-lg"
      data-testid="list-item-test-id"
    >
      <div
        class="relative h-4 w-4 tablet:h-5 tablet:w-5 desktop:h-6 desktop:w-6"
      >
        <img
          alt=""
          src="/icons/cards/check.svg"
        />
      </div>
      Item 2
    </li>
    <li
      class="align-center flex gap-2 text-base tablet:gap-3 tablet:text-lg desktop:gap-4 desktop:text-lg"
      data-testid="list-item-test-id"
    >
      <div
        class="relative h-4 w-4 tablet:h-5 tablet:w-5 desktop:h-6 desktop:w-6"
      >
        <img
          alt=""
          src="/icons/cards/check.svg"
        />
      </div>
      Item 3
    </li>
  </ul>
</DocumentFragment>
`;
